PHP 8.0 support

arueckauer opened the pull request:
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | yes/no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description

This PR provides provides changes described in #12.

Fixes #12

arueckauer committed:
Update supported PHP versions
Upgrade phpunit/phpunit to 9.3

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

arueckauer committed:
Update PHPUnit configuration to best practice

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

arueckauer committed:
Update tests to run with PHPUnit 9.3

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

arueckauer committed:
Add line break before 120 characters as defined by Coding Standard

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

arueckauer committed:
Add PHP 8 support
Drop support for PHP less then 7.3

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

arueckauer committed:
Remove fail on risky and warning configuration to restore previous behavior

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

arueckauer commented:
Hi @boesing 👋🏻

It seems Travis configuration needs some update for PHP nightly builds. I am unsure how to fix that. Has this issue been solved in another component where PHP 8 support has been added?

boesing commented:
Hm, when I understand this correct, its because of the `TESTS_LAMINAS_DIAGNOSTICS_APCU_ENABLED` global environment var. 

We should install `apcu` [5.1.19](https://pecl.php.net/package/APCu/5.1.19) as it is compatible with PHP 8.0.
You can have a look at [laminas-cache-storage-adapter-apcu](https://github.com/laminas/laminas-cache-storage-adapter-apcu/blob/1.1.x/.travis.yml) to see how to install pecl extensions during CI.

arueckauer committed:
Add PHP 8.0 compatible APCu version

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

arueckauer committed:
Fix YAML line indentation

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

arueckauer committed:
Remove command to add config which is irrelevant for this component

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

arueckauer committed:
Disable Memcached for PHP 8.0 since there is not support for it yet

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

arueckauer committed:
Add PHP 8.0 compatible MongoDB version

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

arueckauer commented:
Thanks for the input. APCu now works.

Since Memcached is not yet supported for PHP 8.0 (see latest version [3.1.5](https://pecl.php.net/package/memcached/3.1.5)), I disabled it.

For MongoDB I added the 1.9.0RC1 which adds support for PHP 8. Unfortunately the build was not executed, due to a negative credit balance in Travis.

> Owner laminas does not have enough credits.

boesing commented:
Thanks for the changes. We might want to have a look when travis is back up and running. 👍

arueckauer committed:
Use local XML schema definition

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

arueckauer commented:
@boesing Builds are generally running again, but the latest push did not trigger Travis. Can you manually start the build process?

arueckauer committed:
Move Composer argument --ignore-platform-reqs to nightly builds

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

arueckauer committed:
Update build matrix
Change PHP version to 8.0

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

arueckauer committed:
Upgrade phpunit/phpunit to 9.4

Signed-off-by: Andi Rückauer <arueckauer@gmail.com>

mathieupetrini commented:
Hi, 

https://github.com/bovigo/vfsStream/pull/232 has been merged. 
Is it possible to restart failing jobs ?

samsonasik commented:
restarted

Ocramius commented on phpunit.xml.dist:
```diff
@@ -2,18 +2,23 @@
     <testsuites>
         <testsuite name="laminas-diagnostics">
-            <directory>./test</directory>
+            <directory suffix="Test.php">test</directory>
```
I don't think this suffix is needed, although it's not a problem either

Ocramius commented on test/GuzzleHttpServiceTest.php:
```diff
@@ -99,17 +99,19 @@ public function testCanSendJsonRequests()
-    public function testCanSendArbitraryRequests()
+    public function testCanSendArbitraryRequests(): void
     {
+        self::markTestSkipped('Clarify what to do with assertion of protected property.');
```
To be discussed **before** merging

boesing replied:
Yeah, just use reflection for now as it might test something which has been fixed before.
I dont really like checking protected/private properties at all but this is done in so many packages and I dont get the reason behind that...

Ocramius replied:
Urgh, this went under my radar: OK to merge without it

boesing reviewed (changes requested):
LGTM

Please remove the skipped test for guzzle and use reflection for now to make the test pass.

weierophinney committed:
qa: switch to GHA instead of Travis

- Adds a coding standards workflow
- Adds a unit tests workflow
- Removes Travis configuation

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: use version of vfsstream that supports PHP 8

Unreleased, so having to alias for now.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
fix: fix failing test under PHP 8

warning is no longer raised; type error is raised instead.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney commented:
I've taken the liberty of amending this with changes to the CI pipeline. We're moving off of Travis to GHA, and I plan to provide fixes per the feedback provided already... but we're out of Travis minutes for the month, so first step is getting a working CI pipeline. :smile:

As you can see, all green at this stage, and we have tests against all expected services.

Ocramius commented on composer.json:
```diff
@@ -27,17 +27,18 @@
         "laminas/laminas-coding-standard": "~1.0.0",
         "laminas/laminas-loader": "^2.0",
-        "mikey179/vfsstream": "^1.6",
+        "mikey179/vfsstream": "dev-master#08a798577d677436f4f4ea3d0b3c949f64655548 as 1.6.9",
```
To be changed before merge

Ocramius replied:
Hmm, no, only `require-dev`: I'd say "roll with it"

weierophinney replied:
Until they have a release that includes that commit... it is what it is. :)

Ocramius commented:
> We're moving off of Travis to GHA

@weierophinney the GHA setup used here seems to be a bit wonky, but in the interest of getting anything running at this point, I would say :ship:
